youtube_dl/extractor/kick.py

```diff
@@ -0,0 +1,46 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+
+
+class KickIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?kick\.com/video/(?P<id>[0-9a-zA-Z-]+)'
+    _TEST = {
+        'url': 'https://kick.com/video/82a3c11d-7a17-4747-aecb-2e61413eb11f',
+        'md5': 'f052bc1046cd9ca6751925dd12420010',
+        'info_dict': {
+            'id': '82a3c11d-7a17-4747-aecb-2e61413eb11f',
+            'ext': 'm3u8',
+            'title': 'Weekly Stake Stream',
+            'uploader': 'Eddie',
+            'thumbnail': r're:^https?://.*\.jpg.*$',
+        }
+    }
+
+    def _real_extract(self, url):
+        id = self._match_id(url)
+
+        headers = {
+            'Accept': 'application/json',
+            'User-Agent': 'Mozilla',
+        }
+
+        data = self._download_json('https://kick.com/api/v1/video/%s' % id, id, headers=headers)
+
+        video_url = data['source']
+        formats = self._extract_m3u8_formats(
+            data['source'], id, 'mp4')
+        self._sort_formats(formats)
+        title = data['livestream']['session_title']
+        uploader = data['livestream']['channel']['user']['username']
+        thumbnail = data['livestream']['thumbnail']
+
+        return {
+            'url': video_url,
+            'id': id,
+            'title': title,
+            'uploader': uploader,
+            'thumbnail': thumbnail,
+            'formats': formats,
+        }
```

Comment:
Also, from the example I posted in the linked issue:
  
```suggestion
            'timestamp': traverse_obj(data, 'updated_at', 'created_at', expected_type=parse_iso8601),
            'release_timestamp': parse_iso8601(data.get('created_at')),
            'view_count': int_or_none(data.get('views')),
            'channel_id': strip_or_none(data['livestream'].get('channel_id')),
            'is_live': data['livestream'].get('is_live'),
            'duration': float_or_none(data['livestream'].get('duration'), scale=1000),
            'channel': traverse_obj(data['livestream'], ('channel', 'slug'), expected_type=lambda x: strip_or_none(x) or None),
            'uploader': traverse_obj(data['livestream'], ('channel', 'user', 'username'), expected_type=lambda x: strip_or_none(x) or None),
            'categories': traverse_obj(data, ('categories', Ellipsis, 'name'), expected_type=lambda x: strip_or_none(x) or None) or None,
            'tags': traverse_obj(data, ('categories', Ellipsis, 'tags', Ellipsis), expected_type=lambda x: strip_or_none(x) or None) or None,
        }
```
What do these `livestream` fields mean, though?
* `'source': 'obs',`
* `'playback_url': 'https://fa723fc1b171.us-west-2.playback.live-video.net/api/video/v1/us-west-2.196233775518.channel.xJf2jIQZk382.m3u8',`

Reply:
for ```source``` I assume it's the recording software used ([OBS ](https://obsproject.com/)is a popular choice for streamers)

for ```playback_url``` I'm not sure but my theory is that it's the url used _while_ the stream is happening (I'm just looking at the live-video hostname)